### PR TITLE
Make tests working

### PR DESCRIPTION
**Pull request by rilian-la-te**
Unit tests now works and passes AFAIK.

There are some changes in non-testing areas, please, check.

### CODE REVIEW

**Comment by rilian-la-te** [comment]
Will look for running without H3 data now.

File: test/entity/CArtifactTest.cpp [hunk 1]
```diff
@@ -36,7 +36,10 @@ TEST_F(CArtifactTest, RegistersIcons)
 	subject->image = "Test1";
 	subject->large = "Test2";
 
-	auto cb = std::bind(&CArtifactTest::registarCb, this, _1, _2, _3, _4);
+	auto cb = [this](auto && PH1, auto && PH2, auto && PH3, auto && PH4)
```
Comment (dydzio0614) [inline test/entity/CArtifactTest.cpp:39]: I can sense automatic conversion into lambda, which is crap and pretty much always adds needless && and std::forward<decltype> for me

Comment (IvanSavenko) [reply test/entity/CArtifactTest.cpp:39]: But with these changes lambda will run 1 nanosecond faster!

Comment (rilian-la-te) [reply test/entity/CArtifactTest.cpp:39]: This auto conversion was one conversion which I actually ran. It is not for modernization, it just crashes there and was made to easier debugging.

Comment (rilian-la-te) [reply test/entity/CArtifactTest.cpp:39]: Yes, but in command line (in gdb too)

**Comment by rilian-la-te** [comment]
Merged via #2070

**Pull request closed without merging**